### DISCUSSION TITLE
DragDrop/Resizable keep on Grid (int)

### DISCUSSION THREAD

**Discussion by rudacs** [GrapesJS/grapesjs#4895] [Q&A]
Hello,

I'm using absolute mode and I want the components to remain on the grid, which can be 10x10, for example, depending on the user's configuration.

So when moving it would have to keep in this grid 10, 20, 30, 40...
Same thing for the size.

The idea is to allow the user to create a dashboard, but keeping it consistent on the grid.

Any suggestion?

**Comment by artf** [comment]
Unfortunately, there is no such thing at the moment.

**Comment by rudacs** [comment]
I managed to do it on DragDrop.

```
var gridSize = 10;

function round(x, r) {
    return Math.ceil(x / r) * r;
}

function dragGrid(event) {
    var target = event.target;

    var x = target.getStyle('left') || "0";
    var y = target.getStyle('top') || "0";

    x = round(parseFloat(x.replace("px", "")), gridSize);
    y = round(parseFloat(y.replace("px", "")), gridSize);

    event.target.addStyle({ 'left': x + 'px', 'top': y + 'px' });
}

editor.on('component:drag:start component:drag component:drag:end', dragGrid);
```

Still need to see resizable and validate in styleManager
